docs/01-app/03-api-reference/05-config/02-typescript.mdx

```diff
@@ -150,20 +150,44 @@ Next.js will generate a link definition in `.next/types` that contains informati
 ```
 {/* prettier-ignore-end */}
 
-Currently, experimental support includes any string literal, including dynamic segments. For non-literal strings, you currently need to manually cast the `href` with `as Route`:
+Currently, support includes any string literal, including dynamic segments. For non-literal strings, you need to manually cast with `as Route`. The example below shows both `next/link` and `next/navigation` usage:
 
-```tsx
-import type { Route } from 'next';
+```tsx filename="app/example-client.tsx"
+'use client'
+
+import type { Route } from 'next'
 import Link from 'next/link'
+import { useRouter } from 'next/navigation'
 
-// No TypeScript errors if href is a valid route
-<Link href="/about" />
-<Link href="/blog/nextjs" />
-<Link href={`/blog/${slug}`} />
-<Link href={('/blog' + slug) as Route} />
+export default function Example() {
+  const router = useRouter()
+  const slug = 'nextjs'
 
-// TypeScript errors if href is not a valid route
-<Link href="/aboot" />
+  return (
+    <>
+      {/* Link: literal and dynamic */}
+      <Link href="/about" />
+      <Link href={`/blog/${slug}`} />
+      <Link href={('/blog' + slug) as Route} />
```

Comment:
missing slash?

Reply:
Ah - yeah also in line 180 - I vim motioned my way into this using the previous content 😅 

PR's are always welcomed!